### ISSUE TITLE
Trouble installing in Node10 on WSL

### ISSUE THREAD

**Issue by papagunit** [stevetarver/excel-as-json#34]
Currently running windows subsystem for linux, recently installed node 10, here's the output error I get.

```
devon@DESKTOP-HCS87SE:~$ sudo -i
root@DESKTOP-HCS87SE:~# npm install excel-as-json --save-dev
npm WARN deprecated graceful-fs@3.0.11: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js

> libxmljs@0.18.8 install /root/node_modules/libxmljs
> node-pre-gyp install --fallback-to-build --loglevel http

sh: 1: node-pre-gyp: Permission denied
npm WARN enoent ENOENT: no such file or directory, open '/root/package.json'
npm WARN root No description
npm WARN root No repository field.
npm WARN root No README data
npm WARN root No license field.

npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! libxmljs@0.18.8 install: `node-pre-gyp install --fallback-to-build --loglevel http`
npm ERR! spawn ENOENT
npm ERR!
npm ERR! Failed at the libxmljs@0.18.8 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2018-06-19T23_51_04_378Z-debug.log
```

**Comment by stevetarver** [comment]
papgunit: I don't work on the `libxmljs` package - perhaps try asking with that maintainer? I would simplify the install case if you do, to just including `libxmljs` package.

I will leave this open as a tracking item, so other users know there is currently an issue installing on this system.

**Comment by stevetarver** [comment]
TODO: libxml and excel packages have been updated - install updates, compare performance, see if this fixes the above bug. Target Node.js v8.11.4 (LTS) -> v10.9

**Comment by papagunit** [comment]
Believe it or not, some issue with `node-pre-gyp` that `libxmljs` depends on. [This comment](https://github.com/jansmolders86/mediacenterjs/issues/191#issuecomment-317437604) fixed it.

**Comment by jehicks2** [comment]
Attempting to install all dependencies from an existing project freshly cloned and this is the only one that is causing an error. I get similar errors with libxmljs and node-pre-gyp. It's crazy how many of the provided answers on those posts don't solve the issue. 

I get an error of: 

```
#npm install excel-as-json
/
> libxmljs@0.18.8 install C:\Users\James1.Hicks\Documents\VSCode\webpack\competitive-analysis\node_modules\excel-as-json\node_modules\excel\node_modules\libxmljs
> node-pre-gyp install --fallback-to-build --loglevel http

node-pre-gyp http GET https://github.com/libxmljs/libxmljs/releases/download/v0.18.8/node-v46-win32-x64.tar.gz
node-pre-gyp http 400 https://github.com/libxmljs/libxmljs/releases/download/v0.18.8/node-v46-win32-x64.tar.gz
node-pre-gyp ERR! Tried to download(400): https://github.com/libxmljs/libxmljs/releases/download/v0.18.8/node-v46-win32-x64.tar.gz
node-pre-gyp ERR! Pre-built binaries not found for libxmljs@0.18.8 and node@4.7.3 (node-v46 ABI, unknown) (falling back to source compile with node-gyp)
node-pre-gyp http 400 status code downloading tarball https://github.com/libxmljs/libxmljs/releases/download/v0.18.8/node-v46-win32-x64.tar.gz
```

and a whole bunch of other error output after that essentially telling me it is failing with node-pre-gyp and libxmljs. 

First time using a windows and so far have had a horrendous time getting all previous projects to successfully npm install. 

Any idea why this would still be causing issues?

**Comment by stevetarver** [comment]
The snippet you include is curious - 400 = BAD_REQUEST. I just downloaded the listed URL without problem - perhaps a intermittent failure?

Building node gyp is a common windows challenge. Once you get that going, the rest should fall into place.

**Comment by jehicks2** [comment]
Hm...I’ll look at my corporate proxy again about the 400.

Any tips on getting node gyp to work/run/setup? I swear it feels like I’ve tried 20 things.

**Comment by stevetarver** [comment]
Nope, sorry. I don't have the ability to debug Windows issues. I would just have to google and debug...

**Comment by NaveenChahar** [comment]
npm WARN deprecated graceful-fs@3.0.11: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js

> libxmljs@0.18.8 install C:\Microsoft VS Code\Backend\node_modules\excel\node_modules\libxmljs
> node-pre-gyp install --fallback-to-build --loglevel http

node-pre-gyp http GET https://github.com/libxmljs/libxmljs/releases/download/v0.18.8/node-v64-win32-x64.tar.gz
node-pre-gyp http 404 https://github.com/libxmljs/libxmljs/releases/download/v0.18.8/node-v64-win32-x64.tar.gz
node-pre-gyp ERR! Tried to download(404): https://github.com/libxmljs/libxmljs/releases/download/v0.18.8/node-v64-win32-x64.tar.gz
node-pre-gyp ERR! Pre-built binaries not found for libxmljs@0.18.8 and node@10.5.0 (node-v64 ABI, unknown) (falling back to source compile with node-gyp)
node-pre-gyp http 404 status code downloading tarball https://github.com/libxmljs/libxmljs/releases/download/v0.18.8/node-v64-win32-x64.tar.gz


i am getting this error while installing

**Comment by Mayank-jain34** [comment]
Hi, is there any fix for this issue?

**Comment by papagunit** [comment]
@Mayank-jain34 did [this](https://github.com/jansmolders86/mediacenterjs/issues/191#issuecomment-317437604) not work for you? It's possible so many things have changed since, and I haven't used this tool since.

**Comment by Mayank-jain34** [comment]
@papagunit, unfortunately didn't.
